fix(logging): Replace direct stderr/stdout usage in backend_logger with ILogger interface

## What

The `database/utils/backend_logger.h` (lines 91-115) writes log messages directly to `std::cerr` and `std::cout` instead of routing through the `common::interfaces::ILogger` interface from `common_system`. This bypasses the ecosystem's structured logging infrastructure.

```cpp
// Current (incorrect):
std::cerr << "[ERROR] " << message << std::endl;
std::cout << "[INFO] " << message << std::endl;

// Expected:
logger_->error(message);
logger_->info(message);
```

## Why

- **Ecosystem pattern violation**: All other ecosystem projects use `ILogger` from `common_system` for log output
- **Log routing broken**: In production deployments using `logger_system` with file writers, JSON formatting, or remote log aggregation, `backend_logger` output goes to terminal only
- **Monitoring gap**: `monitoring_system` cannot track database log metrics if logs bypass the monitoring-integrated logger
- **Thread safety**: Direct `std::cerr` writes are not synchronized with the async logging pipeline, causing interleaved output

## Where

- `database/utils/backend_logger.h:91-115` — direct `std::cerr`/`std::cout` writes
- `database/core/database_backend.h` — `ILogger` interface is available but not used by backend_logger

## How

1. Accept `std::shared_ptr<common::interfaces::ILogger>` in `backend_logger` constructor
2. Replace `std::cerr <<` with `logger_->error()`
3. Replace `std::cout <<` with `logger_->info()` / `logger_->debug()`
4. Provide a fallback to `std::cerr` only when no logger is injected (during early initialization)
5. Update all `backend_logger` construction sites to pass the logger from the DI context

## Acceptance Criteria

- [ ] `backend_logger` uses `ILogger` interface for all log output
- [ ] Fallback to `std::cerr` exists only when `ILogger` is not available
- [ ] Database log messages appear in `logger_system` output when integrated
- [ ] No direct `std::cout` usage remains in production code
- [ ] Thread safety verified: no interleaved log output under concurrent database operations

--- a/database/utils/backend_logger.h
+++ b/database/utils/backend_logger.h
@@ -41,9 +41,7 @@
 
 #pragma once
 
-#include <kcenon/database/config/feature_flags.h>
-
-#if KCENON_HAS_COMMON_SYSTEM
+#ifdef BUILD_WITH_COMMON_SYSTEM
 #include <kcenon/common/logging/log_functions.h>
 #else
 #include <iostream>
@@ -99,7 +97,7 @@ public:
 	{
 		std::string formatted
 			= "[" + backend_name_ + ":" + std::string(context) + "] " + std::string(message);
-#if KCENON_HAS_COMMON_SYSTEM
+#ifdef BUILD_WITH_COMMON_SYSTEM
 		kcenon::common::logging::log_error(formatted);
 #else
 		std::cerr << formatted << std::endl;
@@ -113,7 +111,7 @@ public:
 	void warning(std::string_view message) const
 	{
 		std::string formatted = "[" + backend_name_ + "] " + std::string(message);
-#if KCENON_HAS_COMMON_SYSTEM
+#ifdef BUILD_WITH_COMMON_SYSTEM
 		kcenon::common::logging::log_warning(formatted);
 #else
 		std::cerr << formatted << std::endl;
@@ -127,7 +125,7 @@ public:
 	void info(std::string_view message) const
 	{
 		std::string formatted = "[" + backend_name_ + "] " + std::string(message);
-#if KCENON_HAS_COMMON_SYSTEM
+#ifdef BUILD_WITH_COMMON_SYSTEM
 		kcenon::common::logging::log_info(formatted);
 #else
 		std::cout << formatted << std::endl;

